### ISSUE TITLE
llvm/asm: missing files

### ISSUE THREAD

**Issue by sbinet** [llir/llvm#20]
hi there,

trying to compile `"llvm/asm"` I get:

```sh
$> go get -u -v github.com/llir/llvm/asm
github.com/llir/llvm (download)
github.com/pkg/errors (download)
package github.com/llir/llvm/asm/internal/lexer: cannot find package "github.com/llir/llvm/asm/internal/lexer" in any of:
	/usr/lib/go/src/github.com/llir/llvm/asm/internal/lexer (from $GOROOT)
	/home/binet/work/igo/src/github.com/llir/llvm/asm/internal/lexer (from $GOPATH)
```

it seems to me there are a few files missing in, at least, `"llvm/asm/internal/{lexer,parser}"`.

could this be fixed?

thx!

**Comment by sbinet** [comment]
to be clear: I understand these files are generated with `gocc`, but:
- it would be easier to incorporate them in git (so the casual user wouldn't have to install `gocc` and regenerate them)
- it would allow `llir/llvm/...` to be go-get-able

one could also create a `go:generate` directive to automate all of this.

(I can send a PR for the above)

**Comment by mewmew** [comment]
Hi Sebastien,

Glad to hear from you. Hope you had a great winter holidays : )

You are correctly pointing out that these files need to be generated at the current stage of the project.

The intention is to check them into git once the project matures, so that `llir/llvm/...` becomes go get:able.

The reason the files have not yet been included yet is simply to limit noise and decrease the download size of the repository as the gocc generated Go source files change a lot each time the [BNF grammar for LLVM IR](https://github.com/llir/llvm/blob/master/asm/internal/ll.bnf) is updated.

> one could also create a go:generate directive to automate all of this.

The Makefile could indeed be replaced by `go:generate` directives, and that works just for me. It would however not make the package go get:able, as `go get` does not run `go generate` and this is by design. Using `go:generate` directives may still make it easier for developers and simplify the installation instructions during the current development stage.

As mentioned above, once the library matures, the generated source files will definitely be checked into the git repository to make the `llir/llvm` project go get:able.

If you do wish to send a PR for the `go:generate` approach, feel free : ) I'd be happy to include it, as long as it doesn't do anything too magical : )

Cheers /u

**Comment by mewmew** [comment]
Marked this issue with the `v0.2` milestone, so that we may include the generated Go source files for each release of `llir/llvm`.

**Comment by mewmew** [comment]
Hi @sbinet!

Just wanted to let you know that llir/llvm is steadily moving closer to a v0.2 release, at which point the source files for the generated parser will be checked into the repo; thus making llir/llvm go:getable : )

More specifically, the v0.2 release will happen as soon as the remaining items on the todo list for read support have been implemented.

https://github.com/llir/llvm/issues/15#issuecomment-290955212

Cheers /u & i

**Comment by mewmew** [comment]
@sbinet Just to let you know. We are preparing the v0.2 release of llir/llvm, and as such the project should be go-getable within the next few days :)

Cheers /u & i

**Comment by mewmew** [comment]
Fixed in rev 6fcd246015d0b438571b0f15e8777659403189bf.

@sbinet The llir/llvm packages are now go-getable :)
